### PR TITLE
Add placement generation for policy sets

### PR DESCRIPTION
**Pull request by ycao56**
https://github.com/stolostron/backlog/issues/19413
Refactor `createPlacement` and `createPlacementBinding` methods to
generate placement and placement binding when policy sets are defined.

Signed-off-by: Yu Cao <ycao@redhat.com>

### CODE REVIEW

File: internal/plugin.go [hunk 1]
```diff
@@ -114,65 +114,92 @@ func (p *Plugin) Generate() ([]byte, error) {
 		}
 	}
 
-	// Keep track of which placement maps to which policy. This will be used to determine
+	// Keep track of which placement maps to which policy and policySet. This will be used to determine
 	// how many placement bindings are required since one binding per placement is required.
-	plcNameToPolicyIdxs := map[string][]int{}
+	// plcNameToPolicyAndSetIdxs[plcName]["policy"] stores the index of policy
+	// plcNameToPolicyAndSetIdxs[plcsetName]["policyset"] stores the index of policyset
```
Comment (mprahl) [inline internal/plugin.go:120]: What if the policy and policy set have the same name?

File: internal/plugin.go [hunk 2]
```diff
@@ -880,13 +908,13 @@ func (p *Plugin) createPlacement(policyConf *types.PolicyConfig) (
 	return
 }
 
-// createPlacementBinding creates a placement binding for the input placement and policies by
+// createPlacementBinding creates a placement binding for the input placement, policies and policy seets by
```
Comment (mprahl) [inline internal/plugin.go:911]:
```suggestion
// createPlacementBinding creates a placement binding for the input placement, policies and policy sets by
```

File: internal/plugin.go (hunk 1 above)
Comment (ycao56) [reply internal/plugin.go:120]: This is a typo. It should be plcNameToPolicyAndSetIdxs[plc~~set~~Name]["policyset"]. Both array will be accounted when placement is being generated on line 160

**Comment by mprahl** [comment]
@ycao56 let me know when you've fixed the typos.

**Review by mprahl — APPROVED** [review]
Nice!

**Pull request merged**